### DISCUSSION TITLE
GA of Acmebot v4 Now Available

### DISCUSSION THREAD

**Discussion by shibayan** [polymind-inc/acmebot#500] [Announcements]
General Availability of Acmebot v4 is now available with upgrades and new features to Azure Functions v4.

Please refer to the GitHub Release for a list of improvements and fixes included in Acmebot v4.
https://github.com/shibayan/keyvault-acmebot/releases/tag/v4.0.0

### Getting started

Upgrading to Acmebot v4 is completed by simply changing two values in the App Settings of Azure Functions.

- `FUNCTIONS_EXTENSION_VERSION` = `~4`
- `WEBSITE_RUN_FROM_PACKAGE` = `https://stacmebotprod.blob.core.windows.net/keyvault-acmebot/v4/latest.zip`

### Feedback

If you find a defect, please create a Discussion or Issue that includes conditions and instructions. It will help me to fix the problem.

**Comment by shibayan** [comment]
I will be releasing the GA version of Acmebot v4 on 8/31.

I strongly recommend that you update your Acmebot v3 to v4 by December 13, 2022, the EOL date for .NET Core 3.1 on which Azure Functions v3 is based.

**Reply by GeorgDangl** [reply]
I just noticed this by accident, because I'm watching this repo. Maybe you could consider to send out a message during renewal, in case users have configured messaging for Teams or Slack? Otherwise, I can image lot's of people missing this, since Acmebot usually runs without any intervention needed😀

**Reply by shibayan** [reply]
Thanks for the feedback. I am looking into ways to encourage people to upgrade as well, but it is difficult to send a notification in a timely manner.

At the very least, we would like to send out an upgrade notice in the Web UI.

**Comment by HenkAJV** [comment]
**Note**: I redeployed the solution and it is working. I am leaving the information below in case it is something that is specific to the upgrade process.

I followed the steps to upgrade. However, I am now seeing the following errors in Azure Portal:

1. **Function App->Configuration->Function Runtime settings tab**: Your app is pinned to un unsupported runtime version fro dotnet. For better performance, we recommend using one of the supported versions instead: ~3

2. **Function App->Overview->Notifications Tab (and top banner)**: System.Private.CoreLib: Could not load file or assembly 'KeyVault.Acmebot, Version=4.0.1.0, Culture=neutral, PublicKeyToken=null'."

Do you have any advice?

Thank you for building this, it's a HUGE timesaver, and make life so much easier.

**Comment by davidjade** [comment]
So I am very late to upgrade (didn't know I could or should). Now that V3 has stopped working for me (the download blob is no longer available it seems), how do I upgrade to V4? Following the instructions above (changing the two environment variables) does not seem to be enough anymore as the new V4 app will not start. I get all sorts of binding endpoint errors. Looking at the newer Azure config, it seems that maybe a bunch of new configuration options are now needed or changed?

Anyone have any ideas? I have certificates expiring in 10 days and I am stuck. Thanks in advance for any help or pointers.

**Reply by shibayan** [reply]
.NET Core 3.1 and Azure Functions v3 are no longer supported. v4 update should be possible by just changing two settings, but you may see an error when viewing the deployed Functions App in Azure Portal. You may see an error when viewing the deployed Functions App in Azure Portal. In some cases, simply restarting the system will fix the problem.

**Comment by HenkAJV** [comment]
Hi,

I ended up reinstalling the entire solution again,  pointing to the key vault with the certificates,  which it picked up,  and process accordingly.

Take snapshots of all your current settings,  so you can reapply after you deploy the latest version.

Henk



Sent from my smartphone. Please excuse the brevity or grammar.



-------- Original message --------
From: David Jade ***@***.***>
Date: 24/03/24 9:54 AM (GMT+12:00)
To: shibayan/keyvault-acmebot ***@***.***>
Cc: Henk Verhoeven ***@***.***>, Comment ***@***.***>
Subject: Re: [shibayan/keyvault-acmebot] GA of Acmebot v4 Now Available (Discussion #500)


So I am very late to upgrade (didn't know I could or should). Now that V3 has stopped working for me (the download blob is no longer available it seems), how do I upgrade to V4? Following the instructions above (changing the two environment variables) does not seem to be enough anymore as the new V4 app will not start. I get all sorts of binding endpoint errors. Looking at the newer Azure config, it seems that maybe a bunch of new configuration options are now needed or changed?

Anyone have any ideas? I have certificates expiring in 10 days and I am stuck. Thanks in advance for any help or pointers.

**Comment by davidjade** [comment]
Well, my issues is without the V3 app working, I cannot get to my configuration anymore. I'm digging through the storage account looking for bits and pieces of configuration but so far no luck.
